### ISSUE TITLE
qmap doesn't work with Iterators.product

### ISSUE THREAD

**Issue by samuela** [tro3/ThreadPools.jl#15]
I'm finding that qmap doesn't work with `Iterators.product`s.
```julia
julia> ThreadPools.qmap(Iterators.product(1:3, 1:3)) do _
           0
       end
ERROR: MethodError: no method matching empty(::Base.Iterators.ProductIterator{Tuple{UnitRange{Int64},UnitRange{Int64}}})
Closest candidates are:
  empty(::BitSet) at bitset.jl:43
  empty(::BitSet, ::Type{Int64}) at bitset.jl:43
  empty(::DataStructures.IntSet) at /home/samuela/.julia/packages/DataStructures/iWRBf/src/int_set.jl:16
  ...
Stacktrace:
 [1] _detect_type(::Function, ::Base.Iterators.ProductIterator{Tuple{UnitRange{Int64},UnitRange{Int64}}}) at /home/samuela/.julia/packages/ThreadPools/KEWRC/src/interface.jl:11
 [2] qmap(::Function, ::Base.Iterators.ProductIterator{Tuple{UnitRange{Int64},UnitRange{Int64}}}) at /home/samuela/.julia/packages/ThreadPools/KEWRC/src/simplefuncs.jl:97
 [3] top-level scope at REPL[53]:1

```
`collect()`ing on the iterator first doesn't seem to work either...

**Comment by tro3** [comment]
Ah - `collect` is the way to go, but qmap is only targeted at 1D arrays.  I will take a look to see if there is a way to convert to multi-dimensional.

**Comment by samuela** [comment]
Is there a way to work directly on iterators? That seems like the ideal resolution imho.

**Comment by tro3** [comment]
It's tricky, but possible.  I'd have to dispatch between Arrays and Iterators (Arrays are going to be faster, I suspect), but I think it is doable.  I'll take a look.

**Comment by tro3** [comment]
Addressed in v1.1.2, which will propagate in the next hour or so. 

```julia
julia> ThreadPools.qmap(Iterators.product(1:3, 1:3)) do _
         0
       end
3×3 Array{Int64,2}:
 0  0  0
 0  0  0
 0  0  0
```
